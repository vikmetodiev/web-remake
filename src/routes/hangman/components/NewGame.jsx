import React from 'react';
import img from '../img/arrow.png';
import styled from 'styled-components';

const Button = styled.button`
  font-size: 24px;
  border: none;
  background: none;
  position: relative;
  cursor: pointer;
  transition: all 2s linear;
  display: ${props => (props.newGame ? 'none' : 'block')};
  opacity: ${props => (props.newGame ? 0 : 1)};
  font-family: 'Neucha', sans-serif;
  &:after {
    content: '';
    width: 36px;
    height: 36px;
    position: absolute;
    background: url(${img}) no-repeat;
    background-size: 100%;
    top: 100%;
    left: 50px;
    border-radius:25px;
  }
  &:hover:after {
    transform: rotate(360deg);
    transition: 600ms linear all;
  }
`;

const NewGame = props => (
  <Button {...props} onClick={props.onClick} style={{color:"grey"}}>
    Another one?
  </Button>
);

export default NewGame;
