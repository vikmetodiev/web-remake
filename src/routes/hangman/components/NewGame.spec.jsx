import React from 'react';

import toJson from 'enzyme-to-json';
import NewGame from './NewGame';


import Adapter from 'enzyme-adapter-react-16';
import { shallow, configure } from 'enzyme';

configure({adapter: new Adapter()});
describe('New Game', () => {
  const item = 'Another one?';

  it('should render the component', () => {
    const wrapper = shallow(
      <NewGame>
        {item}
      </NewGame>);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
