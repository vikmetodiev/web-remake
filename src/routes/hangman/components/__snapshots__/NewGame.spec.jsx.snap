// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`New Game should render the component 1`] = `
<styled.button>
  Another one?
</styled.button>
`;
